engine/access/state_stream/backend/backend_executiondata2_test.go

```diff
@@ -0,0 +1,1065 @@
+		// an identical, properly linked execution result. Otherwise, depending on
+		// receipt order, optimistic sync may consider the fork abandoned.
+		receipt2.ExecutionResult = receipt1.ExecutionResult
+		prevResultID = receipt1.ExecutionResult.ID()
+
+		receipt1.ExecutorID = s.fixedExecutionNodes[0].NodeID
+		receipt2.ExecutorID = s.fixedExecutionNodes[1].NodeID
+
+		// store execution result for this block (shared by both receipts)
+		s.executionResults[i] = &receipt1.ExecutionResult
+
+		receipts := flow.ExecutionReceiptList{receipt1, receipt2}
+		s.receipts.
+			On("ByBlockID", block.ID()).
+			Return(receipts, nil).
+			Maybe()
+	}
+
+	s.snapshot = protocolmock.NewSnapshot(s.T())
+	s.snapshot.On("Head").Return(s.blocks[0].ToHeader(), nil).Maybe()
+	s.snapshot.On("SealedResult").Return(s.executionResults[0], nil, nil).Maybe() // seal is not used
+	s.snapshot.On("Identities", mock.Anything).Return(s.fixedExecutionNodes, nil).Maybe()
+
+	s.params = protocolmock.NewParams(s.T())
+	s.params.On("SporkRootBlockHeight").Return(s.sporkRootBlock.Height, nil).Maybe()
+	s.params.On("SporkRootBlock").Return(s.sporkRootBlock, nil).Maybe()
+
+	s.state = protocolmock.NewState(s.T())
+	s.state.On("AtBlockID", mock.Anything).Return(s.snapshot, nil).Maybe()
+	s.state.On("Sealed").Return(s.snapshot, nil).Maybe()
+	s.state.On("Params").Return(s.params).Maybe()
+	s.state.On("Final").Return(s.snapshot, nil).Maybe()
+
+	s.headers = storagemock.NewHeaders(s.T())
+
+	// execution data stuff
+	s.executionDataBroadcaster = engine.NewBroadcaster()
+	s.executionDataTrackerReal = tracker.NewExecutionDataTracker(
+		s.log,
+		s.state,
+		s.sporkRootBlock.Height,
+		s.headers,
+		s.executionDataBroadcaster,
+		s.sporkRootBlock.Height,
+	)
+
+	s.executionDataTracker = trackermock.NewExecutionDataTracker(s.T())
+
+	s.subscriptionFactory = subscription.NewSubscriptionHandler(
+		s.log,
+		s.executionDataBroadcaster,
+		subscription.DefaultSendTimeout,
+		subscription.DefaultResponseLimit,
+		subscription.DefaultSendBufferSize,
+	)
+
+	// optimistic sync stuff
+	s.executionResultProvider = osyncmock.NewExecutionResultInfoProvider(s.T())
+	s.executionDataSnapshot = osyncmock.NewSnapshot(s.T())
+	s.executionStateCache = osyncmock.NewExecutionStateCache(s.T())
+	s.criteria = optimistic_sync.DefaultCriteria
+
+	executionNodeSelector := execution_result.NewExecutionNodeSelector(
+		s.preferredExecutionNodeIDs,
+		s.fixedExecutionNodes.NodeIDs(),
+	)
+
+	s.executionResultProviderReal = execution_result.NewExecutionResultInfoProvider(
+		s.log,
+		s.state,
+		s.receipts,
+		executionNodeSelector,
+		s.criteria,
+	)
+}
+
+// TestSubscribeExecutionData verifies that the subscription to execution data works correctly
+// starting from the spork root block ID. It ensures that the execution data is received
+// sequentially and matches the expected data.
+func (s *BackendExecutionDataSuite2) TestSubscribeExecutionData() {
+	s.mockDataProviderState()
+
+	s.executionDataTracker.
```

Comment:
in the updated versions from master (maybe just for events?) I removed this mock. I don't think it's used in most cases and adds complexity to the test. I think you should just use the mock in the specific cases where we need it.